packages/fluentui/react-northstar/src/components/Carousel/Carousel.tsx

```diff
@@ -128,104 +137,114 @@ export const carouselSlotClassNames: CarouselSlotClassNames = {
+        handleNextPaddleFocus();
+      },
+      showPreviousSlideByPaddlePress: e => {
+        e.preventDefault();
+        showPreviousSlide(e, false);
+        handlePreviousPaddleFocus();
+      },
+    },
+    mapPropsToBehavior: () => ({
+      navigation,
+      ariaLiveOn,
+      ariaRoleDescription,
+      ariaLabel,
     }),
-    activeIndex: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-    ariaRoleDescription: PropTypes.string,
-    ariaLabel: PropTypes.string,
-    circular: PropTypes.bool,
-    defaultActiveIndex: PropTypes.oneOfType([PropTypes.number, PropTypes.string]),
-    getItemPositionText: PropTypes.func,
-    items: customPropTypes.collectionShorthand,
-    navigation: PropTypes.oneOfType([customPropTypes.collectionShorthand, customPropTypes.itemShorthand]),
-    navigationPosition: PropTypes.string,
-    onActiveIndexChange: PropTypes.func,
-    paddleNext: customPropTypes.itemShorthand,
-    paddlesPosition: PropTypes.string,
-    paddlePrevious: customPropTypes.itemShorthand,
-    thumbnails: PropTypes.bool,
-  };
-
-  static autoControlledProps = ['activeIndex'];
-
-  static defaultProps = {
-    accessibility: carouselBehavior as Accessibility,
-    paddlePrevious: {},
-    paddleNext: {},
-  };
-
-  static Item = CarouselItem;
-  static Navigation = CarouselNavigation;
-  static NavigationItem = CarouselNavigationItem;
-  static Paddle = CarouselPaddle;
-
-  static getAutoControlledStateFromProps(props: CarouselProps, state: CarouselState) {
-    const { items } = props;
-    const { itemIds } = state;
-
-    if (!items) {
-      return null;
-    }
+  });
 
-    return {
-      itemIds: items.map((item, index) => getOrGenerateIdFromShorthand('carousel-item-', item, itemIds[index])),
-    };
-  }
+  const { classes } = useStyles<CarouselStylesProps>(Carousel.displayName, {
+    className: carouselClassName,
+    mapPropsToStyles: () => ({
+      shouldFocusContainer,
+      isFromKeyboard,
+    }),
+    mapPropsToInlineStyles: () => ({
+      className,
+      design,
+      styles,
+      variables,
+    }),
+    rtl: context.rtl,
+  });
 
-  componentWillUnmount() {
-    this.focusItemAtIndex.cancel();
-  }
+  const paddleNextRef = React.createRef<HTMLElement>();
+  const paddlePreviousRef = React.createRef<HTMLElement>();
 
-  actionHandlers = {
-    showNextSlideByKeyboardNavigation: e => {
-      e.preventDefault();
-      this.showNextSlide(e, true);
-    },
-    showPreviousSlideByKeyboardNavigation: e => {
-      e.preventDefault();
-      this.showPreviousSlide(e, true);
-    },
-    showNextSlideByPaddlePress: e => {
-      e.preventDefault();
-      this.showNextSlide(e, false);
-      this.handleNextPaddleFocus();
-    },
-    showPreviousSlideByPaddlePress: e => {
-      e.preventDefault();
-      this.showPreviousSlide(e, false);
-      this.handlePreviousPaddleFocus();
-    },
-  };
+  const focusItemAtIndex: DebounceResultFn<(index: number) => void> = _.debounce((index: number) => {
```

Comment:
```suggestion
  const focusItemAtIndex = _.debounce((index: number) => {
```

This can be removed as it was done to avoid exposing Lodash types in class methods